### DISCUSSION TITLE
Docking Port wires always locked

### DISCUSSION THREAD

**Discussion by Jav1k** [FakeFishGames/Barotrauma#13856] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Any docking port is completely unable to be rewired, no matter if its on a private game and/or custom sub. I assume this is to reduce griefing in public servers but it hurts gameplay options for engineers in private games. The components that control the hatch on the docking port can still be picked when cutting open the wall, rendering the docking hatch permanently inoperable and unfixable. The best solution to this would be removing the hard coded wire lock on the docking hatch and locking wires and pickablility on hatch and components on vanilla subs, or adding a server option to permanently lock "essential ship functions" that is on by default. Rewiring the docking port is not even the worst grief that can happen that would require a session restart on a public server and it just ends up limiting the freedom of modifying your sub during gameplay in private games. Please consider changing this in some way that allows modification, at least making it optional if you think this is such a huge issue for the average public server.

### Reproduction steps

1. plasma cut the wall to the right of the docking hatch on any vanilla sub
2. take a wrench and remove any of the 3 components there, the attached wires will drop and be non-interactable
3. nobody will ever be able to fix the damage done if the sub docks to/undocks from a station after it happens
4. its quite annying and limiting when youre playing with friends in a private server

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.4.5.0 (Blood in the Water Update, hotfix 1)

### Which operating system did you encounter this bug on?

Windows

**Comment by JeroonBosch** [comment]
Hi, thank you for your report.

I looked into this for a bit, and the docking hatch is only locked while you're in an outpost. This is to prevent grievers messing with the wires and then having the (messed up) wires saved when starting a mission. Head out onto a mission and you technically are able to change the docking hatch wiring (but it is not easy to do). 

We do think this needs to be improved though, and will be working on solution(s).

In the meantime there are some work arounds:
1. Go in a diving suit and swim above the top docking hatch, holding Alt key allows you to select the Docking Hatch and rewire it.
2. Find an abandoned outpost or a beacon station that allows docking with the top hatch. You should be able to rewire the Docking Hatch more easily

As a sidenote, submarines have been updated in the last update, there are no longer any hidden components in the walls (there is a circuitbox, which can be fully accessed, now instead).
